### PR TITLE
Fixed Azure.AppService.AvailabilityZone only detects premium by tier

### LINKED ISSUE #3034: [BUG] Azure.AppService.AvailabilityZone only detects premium by tier property
### Existing rule

Azure.AppService.AvailabilityZone

### Description of the issue

When the plan SKU is deploy via `sku.name` the rule may incorrectly report the plan is not configured for a sone redundant tier.

### Error messages

_No response_

### Reproduction

https://github.com/microsoft/PSRule/issues/1900

https://github.com/tsc-buddy/bicep-registry-modules/blob/ecac86acc0cad980c4a4773c07591a340c312d07/avm/res/web/serverfarm/main.bicep#L130-L133

### Version of PSRule

2.9.0

### Version of PSRule for Azure

1.38.0

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
Fixes #3034

Fixed `Azure.AppService.AvailabilityZone` only detects premium by tier property.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [ ] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
Ready for review @BernieWhite.

We have discussed earlier that it would be nice to have the `AnyOf {}` exit when first condition is met so the reasoning doesn't end up like this, with duplicate. Behavior like the `-or` operator would be highly appreciated.

File: src/PSRule.Rules.Azure/rules/Azure.AppService.Rule.ps1 [hunk 1]
```diff
@@ -201,10 +201,40 @@ Rule 'Azure.AppService.AvailabilityZone' -Ref 'AZR-000442' -Type 'Microsoft.Web/
...
+    AnyOf {
+        $Assert.In($TargetObject, 'sku.name', $sku.name).Reason(
+            $LocalizedData.AppServiceAvailabilityZoneSKU,
+            $TargetObject.name
+        )
+        $Assert.In($TargetObject, 'sku.tier', $sku.tier).Reason(
+            $LocalizedData.AppServiceAvailabilityZoneSKU,
+            $TargetObject.name
+        )
+    }
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.AppService.Rule.ps1:237]: Let's add the paths here for additional detail.

```suggestion
    AnyOf {
        $Assert.In($TargetObject, 'sku.name', $sku.name).ReasonFrom(
            'sku.name',
            $LocalizedData.AppServiceAvailabilityZoneSKU,
            $TargetObject.name
        )
        $Assert.In($TargetObject, 'sku.tier', $sku.tier).ReasonFrom(
            'sku.tier',
            $LocalizedData.AppServiceAvailabilityZoneSKU,
            $TargetObject.name
        )
    }
```

File: tests/PSRule.Rules.Azure.Tests/Azure.AppService.Tests.ps1 [hunk 2]
```diff
@@ -265,6 +265,7 @@ Describe 'Azure.AppService' -Tag 'AppService' {
             $ruleResult.TargetName | Should -Be 'plan-A', 'plan-D';
 
             $ruleResult[0].Reason | Should -BeExactly @(
+                "The app service plan (plan-A) is not deployed with a SKU that supports zone-redundancy."
                 "The app service plan (plan-A) is not deployed with a SKU that supports zone-redundancy."
                 "Path properties.zoneRedundant: Does not exist."
             );
```
Comment (BernieWhite) [inline tests/PSRule.Rules.Azure.Tests/Azure.AppService.Tests.ps1:271]:
```suggestion
            $ruleResult[0].Reason | Should -BeExactly @(
                "Path sku.name: The app service plan (plan-A) is not deployed with a SKU that supports zone-redundancy."
                "Path sku.tier: The app service plan (plan-A) is not deployed with a SKU that supports zone-redundancy."
                "Path properties.zoneRedundant: Does not exist."
            );
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Thanks @BenjaminEngeset. Looks good. Thanks for having a look at this one.

Just a minor suggestion below.

**Comment by BenjaminEngeset** [comment]
Thanks for the great feedback @BernieWhite. I have adjusted accordingly now.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. All good to merge.

**Pull request merged**